lib/Raven/Client.php

```diff
@@ -231,76 +110,67 @@ class Client
     /**
      * @var bool
      */
-    protected $_shutdown_function_has_been_set;
+    protected $_shutdown_function_has_been_set = false;
+
+    /**
+     * @var Configuration The client configuration
+     */
+    protected $config;
 
-    public function __construct($options_or_dsn = null, $options = [])
+    /**
+     * Constructor.
+     *
+     * @param Configuration $config The client configuration
+     */
+    public function __construct(Configuration $config)
```

Comment:
I don\'t approve it. Our users must be use as simple way to construct client as possible. This is a logging library, not a nuclear power station